### DISCUSSION TITLE
Run stepper continuously at constant speed

### DISCUSSION THREAD

**Discussion by jokubasver** [Chr157i4n/PyTmcStepper#75] [Ideas]
For my project, I need to use both position based movement (accelerate, move for one revolution, decelerate, stop), and also continuous, constant speed movement (move indefinitely until told to stop, ideally with acceleration at the start and deceleration at stop, but it's fine without).

As this project shares similarity with AccelStepper, here's an example: https://www.airspayce.com/mikem/arduino/AccelStepper/ConstantSpeed_8pde-example.html

Would be great to have this functionality, or am I missing something and this is already possible?

**Comment by Chr157i4n** [comment]
currently you could use the VACTUAL register of the tmc2209 for getting continous constant speed movement. With Vactual only needed to be set once via UART the RPi4 does not have to do any calculations during movement.
I have written some info about Vactual in the wiki:
https://github.com/Chr157i4n/TMC2209_Raspberry_Pi/wiki/Motor-movement#uartvactual

But i will look into how AccelStepper has this implemented (This library does not only share similarity, it is partly based on AccelStepper).

**Comment by Chr157i4n** [comment]
I have added the function [set_speed](https://github.com/Chr157i4n/TMC2209_Raspberry_Pi/blob/5ca9d5a3075227a00dfae9f24d620c02126a6987/src/TMC_2209/_TMC_2209_move.py#L78C1-L97C24) and [set_speed_fullstep](https://github.com/Chr157i4n/TMC2209_Raspberry_Pi/blob/5ca9d5a3075227a00dfae9f24d620c02126a6987/src/TMC_2209/_TMC_2209_move.py#L100-L106) on the dev branch, so you can use the [run_speed](https://github.com/Chr157i4n/TMC2209_Raspberry_Pi/blob/5ca9d5a3075227a00dfae9f24d620c02126a6987/src/TMC_2209/_TMC_2209_move.py#L409-L427) function like in your example.
I have also added a demo script for this:
https://github.com/Chr157i4n/TMC2209_Raspberry_Pi/blob/dev/demo/demo_script_11_continous_movement.py

**Comment by jokubasver** [comment]
Thank you! I'll test it as soon as I can

**Comment by jokubasver** [comment]
Works as expected! Good to be merged to main branch, IMO.
